### PR TITLE
[Feature] Implement Badge 

### PR DESCRIPTION
**Pull request by minseungseon**
## Description

a Badge component that generates a small badge to a certain position of its child icon. 

<p align="center">

<p align="center">

</p>

## Related Issues
X

## Tests

I added the following tests:

- Simple rendering snapshot tests of Badge props.

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by JeffGuKang** [comment]
@minseungseon You should pass basic test.

File: main/Badge/index.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    opacityVisible = true,
+    variant = 'standard',
+    position = 'right',
+    border,
+    textColor = '#FFFFFF',
+  } = props;
+
+  if (!showZero) {
+    if (count === 0) return null;
+  }
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:84]: New Line before the new paragraph.

File: main/Badge/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+
+## Props  
+  
+|                       | required | types                 | default   |description |
+| :-------------------: | -------- | :-------------------: | :-------: | ---------- |
+| **count**             |          | number                |    1      | Proportionally controls opacity of badge color |
+| **color**             |          | string                | '34AFF9'  | Set badge color |
+| **maximumValue**      |          | number                |     300   | Set maximum limit where opacity stay 1.0, and if count exceeds maximumValue, badge color opacity sets to 0.6 | 
+| **showZero**          |          | boolean               |           | Set whether to show '0' count number in badge or not |
```
Comment (JeffGuKang) [inline main/Badge/README.md]: Does not `showZero` have a default value?

File: .vscode/settings.json [hunk 3]
```diff
@@ -23,5 +23,6 @@
   },
   "javascript.preferences.importModuleSpecifier": "relative",
   "typescript.preferences.importModuleSpecifier": "relative",
+  "editor.formatOnSave": true,
```
Comment (JeffGuKang) [inline .vscode/settings.json]: I agree with the auto lint options. But this is not related code with `Badge` feature. Could you split this to another PR?

File: main/Badge/index.tsx (hunk 1 above)
Comment (minseungseon) [reply main/Badge/index.tsx:84]: resolved it by @meohyun2 !

File: .vscode/settings.json (hunk 3 above)
Comment (minseungseon) [reply .vscode/settings.json]: @JeffGuKang Can you elaborate what you meant by 'not related code with `Badge`' ? our team members seem to be not really familiar with settings.json codes.

File: main/Badge/README.md (hunk 2 above)
Comment (minseungseon) [reply main/Badge/README.md]: > Does not `showZero` have a default value?

working on it, modificiations will be merged soon . thank you!

File: .vscode/settings.json (hunk 3 above)
Comment (JeffGuKang) [reply .vscode/settings.json]: `vscode/settings.json` is not related with Badge feature. Thus, do not import the file in this PR.

File: .vscode/settings.json [hunk 4]
```diff
@@ -23,5 +23,5 @@
   },
   "javascript.preferences.importModuleSpecifier": "relative",
   "typescript.preferences.importModuleSpecifier": "relative",
-  "prettier.configPath": ".prettierrc.js",
-} 
+  "prettier.configPath": ".prettierrc.js"
+}
```
Comment (JeffGuKang) [inline .vscode/settings.json]: Exclude `.vscode/settings.json` from this PR.

Comment (JeffGuKang) [reply .vscode/settings.json]: I restored this file through  [this way](https://stackoverflow.com/questions/2364147/how-to-get-just-one-file-from-another-branch).

File: main/Badge/index.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  margin-right: 3px;
+`;
+
+const StyledDotView = styled.View`
+  position: absolute;
+  top: -5px;
+  ${(props: StyleProps): string => props.position!}: -5px;
+  width: 20px;
+  height: 20px;
+  background-color: ${(props: StyleProps): string => props.color!};
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:56]: same.

File: main/Badge/index.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+  position: absolute;
+  top: -15px;
+  ${(props: StyleProps): string => props.position!}: -10px;
+  width: auto;
+  min-width: 45px;
+  height: 45px;
+  border-color: ${(props: StyleProps): string =>
+    props.border ? props.border : '#00ff0000'};
+  border-width: 3px;
+  background-color: ${(props: StyleProps): string => props.color!};
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:35]: props.color is optional.
Use `background-color: ${(props) => props.color};`

File: main/Badge/index.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+  opacity?: number;
+  position?: string;
+  border?: string;
+  textColor?: string;
+}
+
+const StyledView = styled.View`
+  position: absolute;
+  top: -15px;
+  ${(props: StyleProps): string => props.position!}: -10px;
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:28]: Could you explain about this line?

File: main/Badge/index.tsx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+  min-width: 45px;
+  height: 45px;
+  border-color: ${(props: StyleProps): string =>
+    props.border ? props.border : '#00ff0000'};
+  border-width: 3px;
+  background-color: ${(props: StyleProps): string => props.color!};
+  border-radius: 50px;
+  justify-content: center;
+  align-items: center;
+  opacity: ${(props: StyleProps): number => props.opacity!};
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:39]: `opacity: ${(props) => props.opacity};` is better.

File: main/Badge/index.tsx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+interface StyleProps {
+  color?: string;
+  opacity?: number;
+  position?: string;
+  border?: string;
+  textColor?: string;
+}
+
+const StyledView = styled.View`
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:25]: `const StyledView = styled.View<StyleProps>`

File: main/Badge/index.tsx [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+  border-width: 3px;
+  background-color: ${(props: StyleProps): string => props.color!};
+  border-radius: 50px;
+  justify-content: center;
+  align-items: center;
+  opacity: ${(props: StyleProps): number => props.opacity!};
+`;
+
+const StyledText = styled.Text`
+  color: ${(props: StyleProps): string => props.textColor!};
```
Comment (JeffGuKang) [inline main/Badge/index.tsx:43]: same.

**Review by JeffGuKang — COMMENTED** [review]
I will lint codes related to Badge.

File: .vscode/settings.json (hunk 4 above)
Comment (JeffGuKang) [reply .vscode/settings.json]: `https://stackoverflow.com/questions/2364147/how-to-get-just-one-file-from-another-branch`

**Comment by JeffGuKang** [comment]
Good job guys!!!
See https://github.com/dooboolab/dooboo-ui/pull/334 for how the code updated.

**Comment by JeffGuKang** [comment]
@minseungseon We are using `yarn` instead of `npm`. So `package-lock.json` shouldn't be included in the commit.
